builtin/stash--helper.c

```diff
@@ -625,6 +631,69 @@ static int list_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int show_stat = 1;
+static int show_patch;
+
+static int git_stash_config(const char *var, const char *value, void *cb)
+{
+	if (!strcmp(var, "stash.showStat")) {
+		show_stat = git_config_bool(var, value);
+		return 0;
+	}
+	if (!strcmp(var, "stash.showPatch")) {
+		show_patch = git_config_bool(var, value);
+		return 0;
+	}
+}
+
+static int show_stash(int argc, const char **argv, const char *prefix)
+{
+	int i, ret = 0;
+	struct argv_array args = ARGV_ARRAY_INIT;
+	struct argv_array args_refs = ARGV_ARRAY_INIT;
+	struct stash_info info;
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options,
+			git_stash_helper_show_usage, PARSE_OPT_KEEP_UNKNOWN);
+
+	argv_array_push(&args, "diff");
+
+	for (i = 0; i < argc; ++i) {
+		if (argv[i][0] == '-')
+			argv_array_push(&args, argv[i]);
+		else
+			argv_array_push(&args_refs, argv[i]);
+	}
+
+	if (get_stash_info(&info, args_refs.argc, args_refs.argv)) {
+		argv_array_clear(&args);
+		argv_array_clear(&args_refs);
+		return -1;
+	}
+
+	if (args.argc == 1) {
```

Comment:
Okay, this really would benefit a comment like `/* no flags so far for the `diff` command */` because I stumbled over it again. This `args.argc == 1` condition is just a bit too magic to be self-explanatory.